pkg/volume/iscsi/iscsi_util.go

```diff
@@ -185,14 +192,20 @@ func (util *ISCSIUtil) DetachDisk(c iscsiDiskUnmounter, mntPath string) error {
 		refCount, err := getDevicePrefixRefCount(c.mounter, prefix)
```

Comment:
This seems like a solid solution to me, at least for the moment.  In the long-term, I'm concerned about cases where sessions that exist independently of Kubernetes get logged off because a pod used a volume connected to the target using the same interface (I think this is the same concern @CristianPop cited just now, but I'm not totally sure).  Solving that might be doable with something like resolving all of the symlinks in /dev/disk/by-path to their devices and then checking whether any of the devices that belong to the session are still mounted; I don't know if this would behave well with multipathing, though.

At any rate, this is a better fix than just changing HasPrefix to Contains.